packages/location-management/src/components/LocationUnitView/index.tsx

```diff
@@ -0,0 +1,221 @@
+  return await serve
+    .list({
+      is_jurisdiction: true,
+      return_geometry: false,
+      properties_filter: getFilterParams({ status: 'Active', geographicLevel: 0 }),
+    })
+    .then((response: LocationUnit[]) => response);
+}
+
+/** Parse the hierarchy node into table data
+ *
+ * @param {Array<ParsedHierarchyNode>} hierarchy - hierarchy node to be parsed
+ * @returns {Array<TableData>} array of table data
+ */
+export function parseTableData(hierarchy: ParsedHierarchyNode[]) {
+  const data: TableData[] = [];
+
+  hierarchy.forEach((location, i: number) => {
+    data.push({
+      id: location.id,
+      key: i.toString(),
+      name: location.label,
+      geographicLevel: location.node.attributes.geographicLevel,
+    });
+  });
+  return data;
+}
+
+/** Gets the hierarchy of the location units
+ *
+ * @param {Array<LocationUnit>} location - array of location units to get hierarchy of
+ * @param {string} accessToken - Access token to be used for requests
+ * @returns {Promise<Array<RawOpenSRPHierarchy>>} array of RawOpenSRPHierarchy
+ */
+async function getHierarchy(location: LocationUnit[], accessToken: string) {
+  const hierarchy: RawOpenSRPHierarchy[] = [];
+
+  for await (const loc of location) {
+    const serve = new OpenSRPService(accessToken, API_BASE_URL, LOCATION_HIERARCHY);
+    const data = await serve.read(loc.id).then((response: RawOpenSRPHierarchy) => response);
+    hierarchy.push(data);
+  }
+
+  return hierarchy;
+}
+
+export const LocationUnitView: React.FC = () => {
+  const accessToken = useSelector((state) => getAccessToken(state) as string);
+  const treeData = (useSelector((state) =>
+    getAllHierarchiesArray(state)
+  ) as unknown) as ParsedHierarchyNode[];
+  const dispatch = useDispatch();
+  const currentParentChildren = (useSelector((state) =>
+    getCurrentChildren(state)
+  ) as unknown) as ParsedHierarchyNode[];
+  const [tableData, setTableData] = useState<TableData[]>([]);
+  const [detail, setDetail] = useState<LocationDetailData | 'loading' | null>(null);
+
+  useEffect(() => {
+    if (!treeData.length) {
+      getBaseTreeNode(accessToken)
+        .then((response) => {
+          dispatch(fetchLocationUnits(response));
+          getHierarchy(response, accessToken)
+            .then((hierarchy) => {
+              hierarchy.forEach((hier) => {
+                const processed = generateJurisdictionTree(hier);
+                dispatch(fetchAllHierarchies(processed.model));
+              });
+            })
+            .catch(() => sendErrorNotification('An error occurred'));
+        })
+        .catch(() => sendErrorNotification('An error occurred'));
+    }
+  }, [treeData, accessToken, dispatch]);
+
+  useEffect(() => {
+    const data = parseTableData(currentParentChildren.length ? currentParentChildren : treeData);
+    setTableData(data);
+  }, [treeData, currentParentChildren]);
+
+  if (!tableData.length || !treeData.length) return <Ripple />;
```

Comment:
Let's use `Spin` from Antd instead of `Ripple`

Reply:
done